src/pkg/systemartifact/cleanupcriteria.go

```diff
@@ -0,0 +1,45 @@
+package systemartifact
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/systemartifact/dao"
+	"github.com/goharbor/harbor/src/pkg/systemartifact/model"
+	"time"
+)
+
+var (
+	DefaultCleanupWindowSeconds = 86400
+)
+
+// CleanupCriteria provides an interface that can be implemented
+// by consumers of the system artifact management framework to
+// provide a custom clean-up criteria. This allows producers of the
+// system artifact data to control the lifespan of the generated artifact
+// records and data.
+// Every system data artifact produces must register a cleanup criteria.
+
+type CleanupCriteria interface {
```

Comment:
beside the defaultCriteria, can you give another example about the imple of CleanupCriteria?

Reply:
And for the interface name, I would like to propose to be a straightforward thing, like:

Selector
CandidateSelector

Reply:
An example of a non-default cleanup criteria - consider a harbor component that creates the system artifact record of a particular type that must be valid for 7 days. In such a case, the component would register a custom implementation of the cleanup criteria where-in the `List()` method would return all system artifact records greater than 7 days.
I will add additional methods for filtering for `List()`

Reply:
Renamed interface to `Selector`. Also, added additional interface methods to allow more filters to be specified by the clients.